ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/components/properties/FontSpec.kt

```diff
@@ -110,29 +114,87 @@ internal fun FontSpec.resolve(
-            FontFamily.Monospace.name -> FontSpec.Generic.Monospace
-            else -> getResourceIdentifier(name = info.value, type = "font")
-                .takeUnless { it == 0 }
-                ?.let { fontId -> FontSpec.Resource(id = fontId) }
-                ?: getAssetFontPath(name = info.value)
-                    ?.let { path -> FontSpec.Asset(path = path) }
-                ?: FontSpec.System(name = info.value).also {
-                    Logger.d(
-                        "Could not find a font resource named `${info.value}`. Assuming it's an OEM system font. " +
-                            "If it isn't, make sure the font exists in the `res/font` folder. See for more info: " +
-                            "https://developer.android.com/develop/ui/views/text-and-emoji/fonts-in-xml",
-                    )
-                }
+@Suppress("ReturnCount")
+private fun ResourceProvider.determineFontSpec(fontsConfig: FontsConfig): FontSpec {
+    val bundledFont = getBundledFontSpec(fontsConfig)
+    if (bundledFont != null) {
+        return bundledFont
+    }
+
+    val webFont = getDownloadedFontSpec(fontsConfig)
+    if (webFont != null) {
+        return webFont
+    }
+
+    return FontSpec.System(name = fontsConfig.android.value).also {
+        Logger.d(
+            "Could not find a font resource named `${fontsConfig.android.value}`. Assuming it's an OEM system font. " +
+                "If it isn't, make sure the font exists in the `res/font` folder. See for more info: " +
+                "https://developer.android.com/develop/ui/views/text-and-emoji/fonts-in-xml",
+        )
+    }
+}
+
+@Suppress("NestedBlockDepth")
+private fun ResourceProvider.getBundledFontSpec(
+    fontsConfig: FontsConfig,
+): FontSpec? {
+    val androidInfo = fontsConfig.android
+    val androidInfoValue = androidInfo.value
+    return if (androidInfoValue.isNotEmpty()) {
+        when (androidInfo) {
+            is FontInfo.GoogleFonts -> FontSpec.Google(name = androidInfoValue)
+            is FontInfo.Name -> when (androidInfoValue) {
+                FontFamily.SansSerif.name -> FontSpec.Generic.SansSerif
+                FontFamily.Serif.name -> FontSpec.Generic.Serif
+                FontFamily.Monospace.name -> FontSpec.Generic.Monospace
+                else -> getResourceIdentifier(name = androidInfoValue, type = "font")
+                    .takeIf { it != 0 }
+                    ?.let { fontId -> FontSpec.Resource(id = fontId) }
+                    ?: getAssetFontPath(name = androidInfoValue)
+                        ?.let { path -> FontSpec.Asset(path = path) }
+            }
         }
+    } else {
+        null
     }
+}
+
+private fun ResourceProvider.getDownloadedFontSpec(fontsConfig: FontsConfig): FontSpec.Downloaded? {
+    val webUrl = fontsConfig.web?.value?.takeIf { it.isNotEmpty() }
+    val webHash = fontsConfig.web?.hash?.takeIf { it.isNotEmpty() }
+    if (webUrl != null && webHash == null) {
+        Logger.w(
```

Comment:
Do we want a warning or an error here?

Reply:
Hmm TBH, I feel this is more of a warning... since this could potentially be called for older fonts that haven't been uploaded yet, so they don't have a `web` value. I could potentially refactor this so we only log if `web` is not null though, which might be cleaner. In that case, I could consider making it an error yeah.

Reply:
Ohh right! I didn't realize this was being logged also for when `web` is null. I agree with the approach of only logging if `web` is not null (that's the approach we took in iOS), because in that case, it is an error